File: frontend/src/api/hardwareDaily.ts
import axios from "axios";

import { http } from "./http";

export interface HardwareDailyArticle {
  title: string;
  content_id: string;
  content_text: string;
  url: string;
  comment_count: number;
  vote_up_count: number;
  author_name: string;
  author_profile_url: string;
  author_badge_text: string | null;
  edit_time: string;
  authority_level: string | null;
  thumbnail_url: string | null;
  content_is_excerpt: boolean;
}

export interface HardwareDailyResponse {
  articles: HardwareDailyArticle[];
  cache: { status: "fresh" | "stale"; fetched_at: string; expires_at: string };
}

export async function getHardwareDailies(): Promise<HardwareDailyResponse> {
  return (await http.get<HardwareDailyResponse>("/api/hardware-daily/latest")).data;
}

export function hardwareDailyError(cause: unknown): string {
  if (axios.isAxiosError(cause)) {
    const detail = cause.response?.data?.detail;
    if (typeof detail?.message === "string") return detail.message;
    if (typeof detail === "string") return detail;
  }
  return "暂时无法读取硬件日报，请稍后重新打开此模块。";
}

File: backend/app/schemas/hardware_daily.py
"""Public wire contracts for the signed-in hardware-daily reader."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field, HttpUrl


class HardwareDailyArticle(BaseModel):
    model_config = ConfigDict(extra="forbid")

    title: str = Field(min_length=1)
    content_id: str = Field(min_length=1)
    content_text: str = Field(min_length=1)
    url: HttpUrl
    comment_count: int = Field(ge=0)
    vote_up_count: int = Field(ge=0)
    author_name: str = Field(min_length=1)
    author_profile_url: HttpUrl
    author_badge_text: str | None = None
    edit_time: datetime
    authority_level: str | None = None
    thumbnail_url: HttpUrl | None = None
    content_is_excerpt: bool = True


class HardwareDailyCacheState(BaseModel):
    model_config = ConfigDict(extra="forbid")

    status: Literal["fresh", "stale"]
    fetched_at: datetime
    expires_at: datetime


class HardwareDailyResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    articles: list[HardwareDailyArticle] = Field(min_length=1)
    cache: HardwareDailyCacheState
